src/app.rs

```diff
@@ -67,7 +67,11 @@ impl AppState {
         let task_limit = args.thread_limit.or_else(|| Some(num_cpus::get() - 1));
         let resource_manager =
             ResourceManager::new(args.s3_connection_limit, args.memory_limit, task_limit);
-        let chunk_cache = ChunkCache::new(args);
+        let chunk_cache = if args.use_chunk_cache {
+            Some(ChunkCache::new(args))
```

Comment:
Nit: Could we make the `ChunkCache::new` constructor only take the arguments it needs explicitly like the `ResourceManager::new` call in the line above to avoid coupling the chunk cache initialiser to our CLI arg formats unnecessarily.